AK/MaybeOwned.h

```diff
@@ -57,6 +63,18 @@ class MaybeOwned {
     bool is_owned() const { return m_handle.template has<NonnullOwnPtr<T>>(); }
 
 private:
+    template<typename F>
+    friend class MaybeOwned;
+
+    template<class U, class D>
+    Variant<NonnullOwnPtr<D>, D*>
+    downcast(Variant<NonnullOwnPtr<U>, U*>&& variant)
+    {
+        if (variant.template has<U*>())
+            return variant.template get<U*>();
+        return static_cast<NonnullOwnPtr<T>&&>(move(variant.template get<NonnullOwnPtr<U>>()));
+    }
+
     Variant<NonnullOwnPtr<T>, T*> m_handle;
```

Comment:
It seems that at some point we could really use a

```
template<typename T>
using Handle = Variant<NonnullOwnPtr<T>, T*>;
```

(Review is still pending, this is just something that I thought of randomly.)

Reply:
You mean generally, not in this PR, yes?

(Should maybe be `V<NNOP, WeakPtr>` if we want to make it a whole thing.)

Reply:
It would be nice if at least the type alias is part of this commit, since we are adding two more usages of that type here. However, it's neither urgent nor strictly required, so I'll leave the decision to you.

Making the raw pointer a `WeakPtr` requires making the stored object `RefCounted` anyways, which is something that we can't do anyways.